core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -237,12 +237,13 @@ protected ManifestEntry<DataFile> prepare(ManifestEntry<DataFile> entry) {
     @Override
     protected FileAppender<ManifestEntry<DataFile>> newAppender(
         PartitionSpec spec, OutputFile file) {
-      Schema manifestSchema = V4Metadata.entrySchema(spec.partitionType());
+      Schema manifestSchema = V4Metadata.entrySchema(spec);
       try {
+        Schema schema = spec.schema();
         return InternalData.write(FileFormat.AVRO, file)
             .schema(manifestSchema)
             .named("manifest_entry")
-            .meta("schema", SchemaParser.toJson(spec.schema()))
+            .meta("schema", SchemaParser.toJson(schema))
```

Comment:
We probably don't want to be writing the full schema anymore since it's not really used anywhere.  I would suggest we swap this for the schema id (we can still tie this back to the table schema used at that point).  Putting this in the file footer really increases the size of the file.

Reply:
it actually looks like our ManifestReader depends on `schema` to exist in https://github.com/apache/iceberg/blob/71629c12d2c609ae2473defdb5954e149f9f72e8/core/src/main/java/org/apache/iceberg/ManifestReader.java#L144